src/java/com/puppetlabs/puppetserver/jruby/ScriptingContainer.java

```diff
@@ -0,0 +1,35 @@
+package com.puppetlabs.puppetserver.jruby;
+
+import org.jruby.embed.LocalContextScope;
+import org.jruby.embed.LocalVariableBehavior;
+
+/**
+ * An extension of the JRuby ScriptingContainer class which is
+ * slightly easier to use from Clojure
+ */
+public class ScriptingContainer
+        extends org.jruby.embed.ScriptingContainer {
+    public ScriptingContainer(LocalContextScope scope) {
+        super(scope);
+    }
+    public ScriptingContainer(LocalContextScope scope, LocalVariableBehavior behavior) {
+        super(scope, behavior);
+    }
+
+    /**
+     * This method delegates to a specific signature of #callMethod from the
```

Comment:
I did.  I've run into it several times in the past.  What happens is that Clojure calls the wrong arity of the Java method, which results in the underlying Ruby method being passed a single array as an argument instead of being passed multiple arguments (or vice versa).  That surfaces in Clojure as a JRuby exception about passing the wrong number of arguments to the Ruby method.

I've run into this enough times in the past where I decided it was probably worth just eliminating the issue once and for all.  I tried a few approaches around type-hinting things in Clojure but they were horrible and didn't seem like they were 100% reliable anyway.  This one seems 100% reliable to me... even though I kind of hate it.